### PR TITLE
“leture02:增加DP和DDP的训练介绍”

### PR DESCRIPTION
**Pull request by Dreaming-future**
lecture02:增加DP和DDP的训练介绍

### CODE REVIEW

File: source/第二章/2.3 并行计算简介.md [hunk 1]
```diff
@@ -76,4 +76,152 @@ for image,label in dataloader:
...
+  # nproc_per_node: 这个参数是指你使用这台服务器上面的几张显卡
+  ```
+
+  > 有时候虽然说，可以简单使用DP，但是DDP的效率是比DP高的，所以很多时候单机多卡的情况，我们还是会去使用DDP
+
+### DP 与 DDP 的区别
+
+#### DP 的优势
+
+如果有的话就是简单，一行代码搞定。
```
Comment (NoFish-528) [inline source/第二章/2.3 并行计算简介.md:193]: 这一部分可以写成DP和DDP的总结性的优缺点，把缺点也可以写上。
1. https://blog.csdn.net/ytusdc/article/details/122091284

File: source/第二章/2.3 并行计算简介.md [hunk 2]
```diff
@@ -76,4 +76,151 @@ for image,label in dataloader:
...
+然后在所有和GPU相关代码的前面添加如下代码，如果不写这句代码，所有的进程都默认在你使用`CUDA_VISIBLE_DEVICES`参数设定的0号GPU上面启动
+
+```python
+  torch.cuda.set_device(args.local_rank) # 调整计算的位置
+  ```
+
+设置GPU之间通信使用的后端和端口，也就是初始化进程组：
+
+```python
+  torch.distributed.init_process_group(backend='nccl') # 初始化进程组
```
Comment (NoFish-528) [inline source/第二章/2.3 并行计算简介.md:158]: 除了nccl 可以简单介绍一下其他的，并且介绍下我们为什么使用nccl
nccl backend is currently the fastest and highly recommended backend when using GPUs. This applies to both single-node and multi-node distributed training. （pytorch官网回答）
[backend对比](https://pytorch.org/docs/stable/distributed.html#torch.distributed.init_process_group)

File: source/第二章/2.3 并行计算简介.md (hunk 1 above)
Comment (NoFish-528) [reply source/第二章/2.3 并行计算简介.md:193]: 看看下方这些是否会有帮助
相关链接：
1. https://github.com/jia-zhuang/pytorch-multi-gpu-training
2. https://zhuanlan.zhihu.com/p/86441879

**Comment by NoFish-528** [comment]
@Dreaming-future 可以再进行一些改动

**Review by NoFish-528 — APPROVED** [review]
LGTM
